src/query/sql/src/planner/binder/table.rs

```diff
@@ -594,7 +601,188 @@ impl Binder {
                     .await
             }
             TableReference::Join { .. } => unreachable!(),
+            TableReference::Values {
+                span,
+                values,
+                alias,
+            } => self.bind_values(bind_context, *span, values, alias).await,
+        }
+    }
+
+    #[async_backtrace::framed]
+    pub(crate) async fn bind_values(
+        &mut self,
+        bind_context: &BindContext,
+        span: Span,
+        values: &Vec<Vec<AExpr>>,
+        alias: &Option<TableAlias>,
+    ) -> Result<(SExpr, BindContext)> {
+        if values.is_empty() {
+            return Err(ErrorCode::SemanticError(
+                "Values lists must have at least one row".to_string(),
+            )
+            .set_span(span));
+        }
+        let same_length = values.windows(2).all(|v| v[0].len() == v[1].len());
+        if !same_length {
+            return Err(ErrorCode::SemanticError(
+                "Values lists must all be the same length".to_string(),
+            )
+            .set_span(span));
+        }
+
+        let mut bind_context = BindContext::with_parent(Box::new(bind_context.clone()));
+        let mut scalar_binder = ScalarBinder::new(
+            &mut bind_context,
+            self.ctx.clone(),
+            &self.name_resolution_ctx,
+            self.metadata.clone(),
+            &[],
+            HashMap::new(),
+        );
+
+        let num_rows = values.len();
+        let num_cols = values[0].len();
+
+        let mut names = match alias {
+            Some(alias) => {
+                if alias.columns.len() > num_cols {
+                    return Err(ErrorCode::SemanticError(format!(
+                        "table {} has {} columns available but {} columns specified",
+                        alias.name,
+                        num_cols,
+                        alias.columns.len()
+                    ))
+                    .set_span(span));
+                }
+                alias
+                    .columns
+                    .iter()
+                    .map(|c| c.to_string())
+                    .collect::<Vec<_>>()
+            }
+            None => vec![],
+        };
+        // For columns whose names are not specified, assigns default column names col0, col2, etc.
```

Comment:
we can throw the error if number is not matched.

see: https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-qry-select-values.html


 If table_alias specifies column names, their number must match the number of expressions per tuple.

Reply:
Both `DuckDB` and `PostgreSQL` support automatically assigning column names to columns that don't have a name, so it's best to be compatible with them.